File: GitToolBox/src/main/java/zielu/gittoolbox/ui/blame/BlameEditorLineData.java
package zielu.gittoolbox.ui.blame;

import com.intellij.openapi.editor.LineExtensionInfo;
import com.intellij.openapi.util.Key;
import java.util.List;
import java.util.Objects;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import zielu.gittoolbox.revision.RevisionInfo;

class BlameEditorLineData {
  static final Key<BlameEditorLineData> KEY = new Key<>("GitToolBox-blame-editor-line-info");

  private final List<LineExtensionInfo> lineInfo;
  private final RevisionInfo revisionInfo;

  BlameEditorLineData(@Nullable List<LineExtensionInfo> lineInfo, @NotNull RevisionInfo revisionInfo) {
    this.lineInfo = lineInfo;
    this.revisionInfo = revisionInfo;
  }

  @Nullable
  List<LineExtensionInfo> getLineInfo() {
    return lineInfo;
  }

  boolean isSameRevision(RevisionInfo revisionInfo) {
    return Objects.equals(this.revisionInfo, revisionInfo);
  }
}

File: src/main/kotlin/zielu/gittoolbox/ui/blame/BlameEditorLineData.kt
package zielu.gittoolbox.ui.blame

import com.intellij.openapi.editor.LineExtensionInfo
import com.intellij.openapi.util.Key
import com.intellij.openapi.util.UserDataHolder
import zielu.gittoolbox.revision.RevisionInfo

internal data class BlameEditorLineData(
  val revisionInfo: RevisionInfo,
  val lineInfo: List<LineExtensionInfo>?
) {

  fun isSameRevision(revisionInfo: RevisionInfo): Boolean {
    return this.revisionInfo == revisionInfo
  }

  companion object {
    private val KEY = Key<BlameEditorLineData>("GitToolBox-blame-editor-line-info")

    @JvmStatic
    fun get(userDataHolder: UserDataHolder): BlameEditorLineData? {
      return userDataHolder.getUserData(KEY)
    }

    @JvmStatic
    fun set(userDataHolder: UserDataHolder, editorData: BlameEditorLineData) {
      userDataHolder.putUserData(KEY, editorData)
    }

    @JvmStatic
    fun clear(userDataHolder: UserDataHolder) {
      userDataHolder.putUserData(KEY, null)
    }
  }
}
